shell/platform/windows/flutter_windows_engine.cc

```diff
@@ -143,6 +147,7 @@ bool FlutterWindowsEngine::RunWithEntrypoint(const char* entrypoint) {
       switches.begin(), switches.end(), std::back_inserter(argv),
       [](const std::string& arg) -> const char* { return arg.c_str(); });
 
+#ifndef FLUTTER_WINUWP
```

Comment:
Appols, please can you be more specific?

Reply:
Anything to do here?

Reply:
Sorry I missed this. In particular, I think we should make an abstract TaskRunner class that the current Win32 version inherits from, and with a temporarily-no-op UWP subclass, and then most of the ifdefs about task runners in this file go away (in fact, if we declare a factory method in the abstract class's header, and implement it in each of the subclass files, then we don't need any ifdefs for task runner at all I think.)

Reply:
I have added an abstract taskrunner, made the win32 version inherit and add the winrt version as well (bonus.. it actually works ;-)).  However, I'm not sure how to implement the factory pattern in a way that would eliminate the ifdef from flutter_windows_engine.  Since c++ doesn't support virtual static inheritance i assume you are implying using a template function for the factory?  If yes, that will not eliminate the need for an ifdef at the calling site.  Also, note that there is another ifdef in that file so even if the factory is possible, that file won't be clean.  Moving the ifdefs in question up into the caller will merely add ifdefs to a file that doesn't have any currently.  So not quite sure what to do about the last aspect of the request here.

Reply:
Please reopen if you have more thoughts.

Reply:
I was envisioning a more low-tech solution:
- Declare a freestanding `CreateTaskRunner` function in task_runner.h
- Implement that function twice, once in the Win32 .cc file, and once in the WinRT .cc file

Since only one of the files will be linked, only one implementation will be present.